### PR TITLE
feat: subscription info in preferences

### PR DESCRIPTION
**Pull request by antsgar**
https://app.asana.com/0/1199914526147392/1200845194469719/f

### CODE REVIEW

**Comment by antsgar** [comment]
Dependent on https://github.com/standardnotes/snjs/pull/411

File: app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      this.userSubscription &&
+      this.availableSubscriptions[this.userSubscription.planName]
+    ) {
+      return this.availableSubscriptions[this.userSubscription.planName].name;
+    }
+    return '';
+  }
+
+  public setUserSubscription(subscription: Subscription): void {
+    console.log('set subscription in state', subscription);
```
Comment (vardan-arm) [inline app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx:43]: Please remove this console statement and the one on line 45 :)

File: app/assets/javascripts/preferences/panes/account/subscription/CancelledSubscription.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { observer } from 'mobx-react-lite';
+import { SubscriptionState } from './subscription_state';
+import { Text } from '@/preferences/components';
+import { Button } from '@/components/Button';
+
+type Props = {
+  subscriptionState: SubscriptionState;
+};
+
+export const CancelledSubscription = observer(
```
Comment (vardan-arm) [inline app/assets/javascripts/preferences/panes/account/subscription/CancelledSubscription.tsx:10]: Seems this and `ActiveSubscription` components are very similar, does it make sense to combine them into one component and render respective parts depending on props?

File: app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx (hunk 1 above)
Comment (mobitar) [inline app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx:43]:
```suggestion
```

**Review by mobitar — APPROVED** [review]

File: app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx (hunk 1 above)
Comment (antsgar) [reply app/assets/javascripts/preferences/panes/account/subscription/subscription_state.tsx:43]: Oops! Nice catch, thanks 🙂

File: app/assets/javascripts/preferences/panes/account/subscription/CancelledSubscription.tsx (hunk 2 above)
Comment (antsgar) [reply app/assets/javascripts/preferences/panes/account/subscription/CancelledSubscription.tsx]: Done! Refactored them into one component

**Review by vardan-arm — APPROVED** [review]

**Review by gorjan5sk — APPROVED** [review]

**Pull request merged**